File: discord-bot-python/email_service.py
import re
import aiosmtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
from config import Config

class EmailService:
    def __init__(self):
        self.host = Config.EMAIL_HOST
        self.port = Config.EMAIL_PORT
        self.user = Config.EMAIL_USER
        self.password = Config.EMAIL_PASSWORD
    
    async def send_verification_code(self, email: str, code: str) -> bool:
        """Send a verification code to the specified email."""
        try:
            # Create message
            message = MIMEMultipart('alternative')
            message['Subject'] = 'Your Discord Verification Code'
            message['From'] = f'"Discord 2FA Bot" <{self.user}>'
            message['To'] = email
            
            # Plain text version
            text = f"""Your verification code is: {code}

This code will expire in {Config.CODE_EXPIRATION_MINUTES} minutes.

If you didn't request this code, please ignore this email."""
            
            # HTML version
            html = f"""
            <html>
              <body style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
                <h2>Discord Verification Code</h2>
                <p>Your verification code is:</p>
                <div style="background-color: #f0f0f0; padding: 20px; text-align: center; font-size: 32px; font-weight: bold; letter-spacing: 5px; margin: 20px 0;">
                  {code}
                </div>
                <p>This code will expire in <strong>{Config.CODE_EXPIRATION_MINUTES} minutes</strong>.</p>
                <p style="color: #666; font-size: 14px;">If you didn't request this code, please ignore this email.</p>
              </body>
            </html>
            """
            
            # Attach both versions
            part1 = MIMEText(text, 'plain')
            part2 = MIMEText(html, 'html')
            message.attach(part1)
            message.attach(part2)
            
            # Send email
            await aiosmtplib.send(
                message,
                hostname=self.host,
                port=self.port,
                username=self.user,
                password=self.password,
                start_tls=True,
            )
            
            return True
        except Exception as e:
            print(f'Error sending email: {e}')
            return False
    
    @staticmethod
    def is_valid_edu_email(email: str) -> bool:
        """Check if the email is a valid .edu email address."""
        edu_regex = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.edu$'
        return bool(re.match(edu_regex, email, re.IGNORECASE))

File: discord-bot-nodejs/src/emailService.ts
import nodemailer from 'nodemailer';
import { config } from './config';

export class EmailService {
  private transporter: nodemailer.Transporter;

  constructor() {
    this.transporter = nodemailer.createTransport({
      host: config.email.host,
      port: config.email.port,
      secure: false, // true for 465, false for other ports
      auth: {
        user: config.email.user,
        pass: config.email.password,
      },
    });
  }

  async sendVerificationCode(email: string, code: string): Promise<boolean> {
    try {
      await this.transporter.sendMail({
        from: `"Discord 2FA Bot" <${config.email.user}>`,
        to: email,
        subject: 'Your Discord Verification Code',
        text: `Your verification code is: ${code}\n\nThis code will expire in ${config.codeExpirationMinutes} minutes.\n\nIf you didn't request this code, please ignore this email.`,
        html: `
          <div style="font-family: Arial, sans-serif; max-width: 600px; margin: 0 auto;">
            <h2>Discord Verification Code</h2>
            <p>Your verification code is:</p>
            <div style="background-color: #f0f0f0; padding: 20px; text-align: center; font-size: 32px; font-weight: bold; letter-spacing: 5px; margin: 20px 0;">
              ${code}
            </div>
            <p>This code will expire in <strong>${config.codeExpirationMinutes} minutes</strong>.</p>
            <p style="color: #666; font-size: 14px;">If you didn't request this code, please ignore this email.</p>
          </div>
        `,
      });
      return true;
    } catch (error) {
      console.error('Error sending email:', error);
      return false;
    }
  }

  isValidEduEmail(email: string): boolean {
    const eduRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.edu$/i;
    return eduRegex.test(email);
  }
}
